src/server/internalServer.cpp

```diff
@@ -801,8 +804,9 @@ std::unique_ptr<Response> InternalServer::handle_content(const RequestContext& r
   } catch(kiwix::NoEntry& e) {
     if (m_verbose.load())
       printf("Failed to find %s\n", urlStr.c_str());
-
-    return Response::build_404(*this, request, bookName);
+    searchPattern = urlStr.substr((urlStr.find_last_of('/'))+1); //Using an extra variable to not overcomplicate the line below.
+    details = "Make a full text search for <a href=\""+m_root+"/search?content="+bookName+"&pattern="+searchPattern+"\">"+searchPattern+"</a>";
```

Comment:
This skipped my attention during a previous review. The second `searchPattern` must be URL-decoded and HTML-escaped.

Reply:
@veloman-yunkan I know that I can use `kiwix::urlDecode` for URL-decode but I cannot find a relevant method on `stringTools.h` for HTML-escape. What should I do there?

Reply:
Adding a new function for HTML-escaping should be easy. However, I propose to create a new template for the **"Make a full text search for <a ...> ... \</a>"**, string and delegate the HTML-escaping job to mustache. Then if we need to make that piece of HTML more fancy we will have to edit that template rather than touch the C++ code in two places.